### DISCUSSION TITLE
Spectrogram normalization constants

### DISCUSSION THREAD

**Discussion by roman-vygon** [NVIDIA-NeMo/NeMo#3530] [Q&A] [answered]
Greetings!
In the [microphone streaming notebook](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Online_ASR_Microphone_Demo.ipynb) the preprocessor is initialized with fixed_mean and fixed_std constants. Where do they come from? How to get their values for a different model\preprocessor. If I am using the default ctc_conformer config do they stay the same?

**Comment by titu1994** [comment, marked as answer]
An earlier discussion with some info about the normalization precomputed using Welfords algorithm - https://github.com/NVIDIA/NeMo/issues/1203#issuecomment-697960406


The constants are precomputed for the dataset you wish to evaluate on. We computed on the then mixture of 3-4 datasets so Librispeech might be slightly different. The constants depend on several factors such as preprocessing window and stride so it is different between QuartzNet Jasper and the rest of the models (Citrinet, Conformer, ContextNet and Conformer Transducer)
